src/main/src/main/java/org/geoserver/security/KeyStoreProviderImpl.java

```diff
@@ -274,8 +270,7 @@ public boolean isKeyStorePassword(char[] password) throws IOException{
             // should not happen, see assertActivatedKeyStore
             throw new RuntimeException(e1);
         }
-        FileInputStream fis =
-                new FileInputStream(getFile());
+        InputStream fis = getResource().in();
```

Comment:
@NielsCharlier please refactor to try-with-resources. Shorter. Less error prone. In this case, an Error will leave the InputStream unclosed.